ddl/ddl_api.go

```diff
@@ -1162,6 +1157,20 @@ func (d *ddl) getModifiableColumnJob(ctx context.Context, ident ast.Ident, origi
 	if err := setDefaultAndComment(ctx, newCol, spec.NewColumn.Options); err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	// copy index related options to the new spec
+	indexFlags := col.FieldType.Flag & (mysql.PriKeyFlag | mysql.UniqueKeyFlag | mysql.MultipleKeyFlag)
```

Comment:
If auto_increment flag is missing in MODIFY COLUMN, the flag should be removed from the new column. Thus it is not copied.